webapp/src/app/api/pull-request/[projectName]/route.ts

```diff
@@ -40,50 +37,36 @@ export async function POST(
 
   try {
     syncLock.set(repoPath, true);
-    // TODO: move logic from here to a separate class
-    const lyraConfig = await LyraConfig.readFromDir(repoPath);
-    const options: Partial<SimpleGitOptions> = {
-      baseDir: repoPath,
-      binary: 'git',
-      maxConcurrentProcesses: 1,
-      trimmed: false,
-    };
-    const git: SimpleGit = simpleGit(options);
-    await git.checkout(lyraConfig.baseBranch);
-    await git.pull();
-    const projectConfig = lyraConfig.getProjectConfigByPath(
+    const repoGit = new RepoGit(repoPath);
+    const baseBranch = await repoGit.checkoutBaseAndPull();
+    const langFilePaths = await repoGit.saveLanguageFiles(
       serverProjectConfig.projectPath,
     );
-    const projectStore = await Cache.getProjectStore(projectConfig);
-    const languages = await projectStore.getLanguageData();
-    const langFilePaths = await writeLangFiles(
-      languages,
-      projectConfig.absTranslationsPath,
-    );
-    const status = await git.status();
-    if (status.files.length == 0) {
+
+    if (await repoGit.statusChanged()) {
```

Comment:
Here we check if status is changed and reject the operation if it is changed, previously we rejected the operation if status was _not_ changed.

Reply:
Oops